src/queue/SoundQueue.ts

```diff
@@ -0,0 +1,57 @@
+import config from '../../config/config.json';
+
+import QueueItem from './QueueItem';
+import Util from '../Util';
+
+export default class SoundQueue {
+  private queue: Array<QueueItem>;
+
+  constructor() {
+    this.queue = [];
+  }
+
+  public add(item: QueueItem) {
+    this.queue.push(item);
+  }
+
+  public isStartable() {
+    return this.queue.length === 1;
+  }
+
+  public start() {
+    this.playNext();
+  }
+
+  public clear() {
+    this.queue.length = 0;
```

Comment:
I know this works but I would argue that `this.queue = [];` is easier to understand!